### PR TITLE
Fix copies main ci

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
After #29909

### CODE REVIEW

File: utils/check_copies.py [hunk 1]
```diff
@@ -313,10 +313,12 @@ def split_code_into_blocks(
...
         ).groups()[0]
-    except ValueError:
+    except Exception:
+        start_context = min(start_index-10, 0)
+        end_context = min(end_index+10, len(lines))
         raise ValueError(
-            f"Tried to split a class or function. It did not work. Error comes from line {start_index}: ```\n"
-            + "".join(lines[start_index:end_index])
+            f"Tried to split a class or function. It did not work. Error comes from line {start_index}: \n```\n"
+            + "".join(lines[start_context:end_context])
```
Comment (ArthurZucker) [inline utils/check_copies.py:321]: just adding some context can help a lot!

File: tests/models/qwen2_moe/test_modeling_qwen2_moe.py [hunk 2]
```diff
@@ -504,7 +504,7 @@ def test_flash_attn_2_generate_use_cache(self):
     @require_torch_gpu
     @pytest.mark.flash_attn_test
     @slow
-    def test_flash_attn_2_inference_padding_right(self):
+    def test_flash_attn_2_inference_equivalence_right_padding(self):
```
Comment (ArthurZucker) [inline tests/models/qwen2_moe/test_modeling_qwen2_moe.py:507]: actual fix

File: utils/check_copies.py [hunk 3]
```diff
@@ -602,8 +604,23 @@ def check_codes_match(observed_code: str, theoretical_code: str) -> Optional[int
...
+                    + observed_code_header
+                    + "\n```\n"
+                )
+
+            try:
+                theoretical_name = re_pattern.search(theoretical_code_header).groups()[0]
+            except Exception:
+                raise ValueError(
+                    "Tried to split a class or function. It did not work. Error comes from: \n```\n"
+                    + observed_code_header
```
Comment (ArthurZucker) [inline utils/check_copies.py:621]:
```suggestion
                    + theoretical_code_header
```

**Review by gante — APPROVED** [review]
Thank you for the fix 🙏 (Extra context is great!)

File: utils/check_copies.py [hunk 4]
```diff
@@ -602,8 +604,23 @@ def check_codes_match(observed_code: str, theoretical_code: str) -> Optional[int
     _re_func_match = re.compile(r"def\s+([^\(]+)\(")
     for re_pattern in [_re_class_match, _re_func_match]:
         if re_pattern.match(observed_code_header) is not None:
-            observed_obj_name = re_pattern.search(observed_code_header).groups()[0]
-            theoretical_name = re_pattern.search(theoretical_code_header).groups()[0]
+            try:
+                observed_obj_name = re_pattern.search(observed_code_header).groups()[0]
+            except Exception:
+                raise ValueError(
+                    "Tried to split a class or function. It did not work. Error comes from: \n```\n"
```
Comment (ydshieh) [inline utils/check_copies.py:611]: The error message is a bit confusing here. The `re_pattern` is used to extract the class or function name, not to split the blocks (unlike the above one)

Comment (ArthurZucker) [reply utils/check_copies.py:611]: feel free to update it. 😄

**Pull request merged**